### PR TITLE
Auth backup5

### PR DESCRIPTION
**Pull request by GonzalezEstevan**
Merged with main and setup post request for backend team on successful signup

### CODE REVIEW

File: utils/context/AuthProvider.js [hunk 1]
```diff
@@ -19,13 +20,26 @@ export function AuthProvider({ children }) {
...
+              console.log(res)
+            }).catch(err => {
+              console.error(err.message)
+            })
+        } else {
+          axios.post('/pages/mentees/index.js', body)
+            .then(res => {
+              console.log(res)
+            }).catch(err => {
+              console.error(err.message)
```
Comment (cmjosephs) [inline utils/context/AuthProvider.js]: Switch console.errors to console.warn()

File: utils/context/AuthProvider.js [hunk 2]
```diff
@@ -19,13 +20,26 @@ export function AuthProvider({ children }) {
...
+         body['uid'] = response.user.uid
+        if (body.account_type === 'Mentor') {
+          axios.post('/pages/mentors/index.js', body)
+            .then(res => {
+              console.log(res)
+            }).catch(err => {
+              console.error(err.message)
+            })
+        } else {
+          axios.post('/pages/mentees/index.js', body)
```
Comment (cmjosephs) [inline utils/context/AuthProvider.js]: Did you mean to send these to this route? Thought api route was `/api/*`. Not sure but lmk if that was working properly

File: pages/signup.js [hunk 3]
```diff
@@ -49,12 +56,28 @@ export default function Signup() {
...
+              label="Account Type"
+              value={accountType}
+              className={classes.userInput}
+              sx={{height:30}}
+              onChange={(e) => {handleSignUpFormInput(e.target.value, 'account_type'), setAccountType(e.target.value)}}
+            >
+              <MenuItem value={'Mentor'}>Mentor</MenuItem>
+              <MenuItem value={'Mentee'}>Mentee</MenuItem>
+              </Select>
+              {console.log(signupInfo)}
```
Comment (cmjosephs) [inline pages/signup.js]: Make sure to take out console.logs after you know its working

File: pages/app/my-profile.js [hunk 4]
```diff
@@ -1,9 +1,24 @@
 // Profile View
 import { Image } from 'next/image';
 
-export default function MyProfile() {
+// import components
```
Comment (cmjosephs) [inline pages/app/my-profile.js]: Go ahead and delete the majority of contents for this file. Just match it up with whatever the most recent main branch has

**Review by cmjosephs — COMMENTED** [review]
Couple of nits, but overall good stuff.

**Comment by GonzalezEstevan** [comment]
> Couple of nits, but overall good stuff.

> changed axios request to next js fetch mthod 
Screen Shot 2022-03-22 at 10 46 43 AM

File: pages/signup.js [hunk 5]
```diff
@@ -32,13 +39,16 @@ export default function Signup() {
...
   }
   if (user) {
     router.push('/app/my-courses')
     return null;
   }
-  return (
+
+
+
+  return !loading ? (
```
Comment (cmjosephs) [inline pages/signup.js]: I don't think this is a clean way to handle the loading state. I know we need to fix the styling with the server side rendering but I'm thinking lets leave this a little cleaner short term even though it causes the refresh problem and address that issue in a couple days.

File: utils/context/AuthProvider.js [hunk 6]
```diff
@@ -19,13 +19,28 @@ export function AuthProvider({ children }) {
...
 
-  // console.log(user);
-
-  const signup = (email, password, username) => {
-     createUserWithEmailAndPassword(auth, email, password)
-      .then((response) => {
+  const signup = (body) => {
+    createUserWithEmailAndPassword(auth, body.email, body.password)
+      .then(async (response) => {
+        body['uid'] = response.user.uid
```
Comment (cmjosephs) [inline utils/context/AuthProvider.js:25]: Love how the uid add is being handled here

File: utils/context/AuthProvider.js [hunk 7]
```diff
@@ -19,13 +19,28 @@ export function AuthProvider({ children }) {
...
+              'Content-Type': 'application/json',
+            },
+            body: body,
+          });
+        }
         router.push('/app/my-courses')
-        console.log(response)
       })
       .catch((err) => {
         console.warn('Problem with sign up: ', err.message);
```
Comment (cmjosephs) [inline utils/context/AuthProvider.js:46]: Fine for now but we should probably look into adding in an error page with a redirect for failed sign ups and log ins

File: utils/context/AuthProvider.js [hunk 8]
```diff
@@ -35,12 +50,12 @@ export function AuthProvider({ children }) {
...
-    })
+      .then((response) => {
+        // need to set redirect to /[username]/dashboard
+        // need to set success state
+        router.push('/app/my-courses');
+        console.log(response)
+      })
       .catch((err) => {
         // needs to set an error state
         console.warn('Problem with log in: ', err.message);
```
Comment (cmjosephs) [inline utils/context/AuthProvider.js:61]: See above comment

**Review by cmjosephs — CHANGES_REQUESTED** [review]
Only thing is changing the way you handled loading state. Lets leave how it was before and add a trello bug ticket and tackle it at a later date. On trello ticket include what's causing the bug at a high level

**Comment by GonzalezEstevan** [comment]
@cmjosephs

**Pull request closed without merging**